cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -519,7 +635,6 @@ func (c *machineController) scalableResourceProviderIDs(scalableResource *unstru
 
 		if !found {
 			klog.V(4).Infof("Status.NodeRef of machine %q is currently nil", machine.GetName())
-			providerIDs = append(providerIDs, fmt.Sprintf("%s%s_%s", pendingMachinePrefix, machine.GetNamespace(), machine.GetName()))
```

Comment:
i think we need to keep this line so that we can see the pending machines coming up in the provider. we have logic in the find function to parse out these machines.

Reply:
Good catch--I'm honestly not sure why this was removed. Will fix.